app/models/user.rb

```diff
@@ -139,6 +141,14 @@ def channels
     channels
   end
 
+  def self.find_or_create_for_browse(is_browse_app, mobile)
```

Comment:
Would be great to write this without the need for `is_browse_app`. The model is currently too tightly coupled to the app type.